src/runtime/composables/useApiData.ts

```diff
@@ -19,13 +19,17 @@ type ComputedOptions<T extends Record<string, any>> = {
       : MaybeRef<T[K]>;
 }
 
-export type BaseUseApiDataOptions<T> = Omit<AsyncDataOptions<T>, 'watch'> & {
+export interface APIError<Client extends boolean, T = any> extends NuxtError {
```

Comment:
I don't think this is correct. `NuxtError` is a wrapper around H3's `H3Error`: https://github.com/nuxt/nuxt/blob/0094b569c713f3cfc9b49b07e1af5733dc45da99/packages/nuxt/src/app/composables/error.ts#L9

In the server handler after the `$fetch` call (from ofetch), we throw a `H3Error` with `createError` and pass the `statusCode`, `statusMessage` and the API's response body to the client with that very body: https://github.com/johannschopplich/nuxt-api-party/blob/e5537f78e33f7b80b62f0589333e5ec65b62eebd/src/runtime/server/handler.ts#L77

The client again will init a `$fetch` call to the server route. Thus, both the server and client use `$fetch`, which both throw a `FetchError`. ofetch handles all the Logic and provides us a unified interface, if I'm not mistaken.

Reply:
It ended up being an `H3Error` (technically `NuxtError` because `__nuxt_error` was set.) when I checked it in the template.

Reply:
Since Nuxt API Party behaves similar to Nuxt's `useFetch` – `useAsyncData` + `$fetch` – we use the same error type: https://github.com/nuxt/nuxt/blob/0094b569c713f3cfc9b49b07e1af5733dc45da99/packages/nuxt/src/app/composables/fetch.ts#L40
That's why I don't understand the need for changes. I want to keep the usage as close to Nuxt source as possible. The H3Error should only be thrown on server-side errors.